crates/wasmtime/src/values.rs

```diff
@@ -89,8 +89,17 @@ impl Val {
             Val::I64(_) => ValType::I64,
             Val::F32(_) => ValType::F32,
             Val::F64(_) => ValType::F64,
-            Val::ExternRef(_) => ValType::ExternRef,
-            Val::FuncRef(_) => ValType::FuncRef,
+            Val::ExternRef(_) => ValType::Ref(RefType { // TODO(dhil):
+                                                        // This is a bug. It is not true that every externref is
+                                                        // nullable. Too see why, just consider the instruction [ref.extern]
+                                                        // it returns a non-nullable extern ref.
+                nullable: true,
+                heap_type: HeapType::Extern,
+            }),
```

Comment:
Could the `nullable` field here be filled out based on whether the value is `Some` or `None`?

Reply:
Yep, it is technically possible since it is valid to use a non-nullable reference in place of a nullable reference. However, it does come with a cost -- which I think is inevitable anyways -- which is that we must perform a subtype check where we would previously perform a standard equality `==` check.